SimpleClient/Source/Handler.cpp

```diff
@@ -22,344 +21,417 @@ Handler::Handler()
-		case State::ROLL_MINUS:
-			[[fallthrough]];
-		case State::PITHCH_PLUS:
-			[[fallthrough]];
-		case State::PITHCH_MINUS:
-			[[fallthrough]];
-		case State::YAW_PLUS:
+                std::size_t count = 0;
+
+                bool flag = true;
+
+                _data += '|';
+
+                for (std::size_t i = 0; i < _data.size(); ++i)
+                {
+                    if (_data[i] == '|')
+                    {
+                        const std::string strToParse = _data.substr(prev, i - prev);
+
+                        prev = i + 1;
+
+                        ++count;
+
+                        bool locFlag = true;
+
+                        if (count >= 2)
+                        {
+                            result.points.emplace_back(utils::fromString<RobotData>(strToParse,
+                                locFlag));
+                        }
+
+                        flag &= locFlag;
+                    }
+                }
+
+                if (flag)
+                {
+                    return result;
+                }
+            }
+
+            break;
+        }
+
+		case State::CIRCLIC:
 			[[fallthrough]];
-		case State::YAW_MINUS:
-			if (_data.size() > 1u)
-			{
-				if (const int coefficient = utils::stringToInt(_data.substr(1u)); coefficient > 0)
-				{
-					result.coefficient = coefficient;
-					setCoefficient(coefficient);
-					return result;
-				}
-
-				_state = State::DEFAULT;
-			}
-			else
+		case State::PARTIAL:
+		{
+            const long long num = std::count(_data.begin(), _data.end(), '|');
+            if (_data.size() > 1u && num >= 3)
 			{
-				return result;
-			}
-			break;
-
-		default:
-			assert(false);
+                result.points.reserve(num - 1);
+
+                std::size_t prev = 0;
+
+                std::size_t count = 0;
+
+			    bool flag = true;
+
+                _data += '|';
+
+                for(std::size_t i = 0; i < _data.size(); ++i)
+                {
+                    if(_data[i] == '|')
+                    {
+                        const std::string strToParse = _data.substr(prev, i - prev);
+
+                        prev = i + 1;
+
+                        ++count;
+
+                        bool locFlag = true;
+
+                        if(count == 2)
+                        {
+                            result.numberOfIterations = utils::fromString<int>(strToParse, locFlag);
+                            if(result.numberOfIterations < 1)
+                            {
+                                flag = false;
+                                break;
+                            }
+                        }
+                        if(count >= 3)
+                        {
+                            result.points.emplace_back(utils::fromString<RobotData>(strToParse,
+                                                                                    locFlag));
+                        }
+
+                        flag &= locFlag;
```

Comment:
**(note):** I think it will be good for all, if you will add commentaries to such strings of your code.

Reply:
It's coomment for Danila.